### PR TITLE
deployment/README.md: add maintenance info

### PR DESCRIPTION
**Pull request by gilligan**
Add information on maintenance tasks:
- adding users
- adding environments

Add aws-upload-secrets.nix for uplading credentials
to the aws secrets manager

--------------

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: deployment/README.md [hunk 1]
```diff
@@ -123,3 +123,113 @@ The deployment depends on several credentials which are maintained using the [AW
...
+- The `alpha` environment is deployed to `alpha.iohkdev.io`
+- `testing` is deployed to `testing.iohkdev.io`
+
+Terraform uses different workspaces for each environment which are also separated in the shared state which is stored in a S3 bucket. When entering a nix-shell the respective terraform workbench is chosen automatically.
+
+In order to add a new environment the following steps need to be followed:
+
+**1. Add the environment to [deployment/envs.nix](https://github.com/input-output-hk/plutus/blob/master/deployment/envs.nix)**
+
+In order to add an environment `username` it needs to be added to the attribute set in `deployment/envs.nix` as follows:
```
Comment (merivale) [inline deployment/README.md:191]: A tiny thing, but maybe we could just have "environment" as the placeholder name for the environment. It needn't be a username (and `alpha` and `testing` aren't).

File: deployment/README.md [hunk 2]
```diff
@@ -123,3 +123,113 @@ The deployment depends on several credentials which are maintained using the [AW
...
+    "githubClientId": "<id>",
+    "githubClientSecret": "<secret>",
+    "jwtSignature": "<jwt-token>"
+  }
+}
+```
+Then use `aws-upload-secrets` to submit it:
+
+```shell
+$ nix-shell aws-upload-secrets.nix --argstr env <name> --run "aws-upload-secrets ./file.json"
```
Comment (merivale) [inline deployment/README.md:227]: nice :sunglasses:

**Review by merivale — APPROVED** [review]
LGTM.

File: deployment/README.md [hunk 3]
```diff
@@ -123,3 +123,113 @@ The deployment depends on several credentials which are maintained using the [AW
...
+```
+
+### Adding Environments
+
+Deployments can be performed to different environments. Each environment is a full aws setup with multiple ec2 instances and networking, deployed to different `iohkdev.io` subdomains:
+
+- The `alpha` environment is deployed to `alpha.iohkdev.io`
+- `testing` is deployed to `testing.iohkdev.io`
+
+Terraform uses different workspaces for each environment which are also separated in the shared state which is stored in a S3 bucket. When entering a nix-shell the respective terraform workbench is chosen automatically.
```
[the remark this answers is not in the document]
Comment (gilligan) [reply deployment/README.md:185]: "workbench" ? *LOL* - no that should be workspace :)

File: deployment/README.md [hunk 4]
```diff
@@ -123,3 +123,143 @@ The deployment depends on several credentials which are maintained using the [AW
...
+$ echo $TF_VAR_plutus_github_client_id # should print the value you just configured
+```
+
+
+### Extending The Deployment
+
+The deployment process is split between provisioning the infrastructure on AWS using Terraform and deploying NixOS configurations with updated packages or service descriptions using morph. Depending on the respective changes, either one or both of these layers have to be updated.
+
+#### Adding Servers
+The currently configured ec2 instances are easy two discover:
```
Comment (merivale) [inline deployment/README.md:243]:
```suggestion
The currently configured ec2 instances are easy to discover:
```

**Review by merivale — APPROVED** [review]

**Pull request merged**